### PR TITLE
Elvis/tab bar

### PR DESCRIPTION
**Pull request by emarc0314**
## Overview

- Added new Activites Page
- Reinstated Tab Bar

## Changes Made

- I added a new page for the Tab Bar for our activites, and included the list of activities I had already defined for our Home page.

## Test Coverage

- Launched on iPhone 13, 14, 14 Pro.

## Next Steps (delete if not applicable)

- Looking to tie the activity cells on the ActivityViewController Page to an ActivityDetailPage to show where that activity can be played, how much it costs to play, etc.

## Screenshots (delete if not applicable)

<details>

  <summary>Home Page and Activity Page</summary>

  
  

</details>

### CODE REVIEW

File: Uplift/Views/Cells/ActivityCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//
+//  Created by Elvis Marcelo on 4/26/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+class ActivityCollectionViewCell: UICollectionViewCell {
+
+    let iconImageView = UIImageView()
```
Comment (ReadeP) [inline Uplift/Views/Cells/ActivityCollectionViewCell.swift:13]: these should be private

File: Uplift/Controllers/ActivitiesViewController.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+
+        setupViews()
+        setupConstraints()
+    }
+
+    func setupViews() {
```
Comment (ReadeP) [inline Uplift/Controllers/ActivitiesViewController.swift:29]: should be private

File: Uplift/Views/Cells/ActivityCollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    var activity: Activity!
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupViews()
+        setupConstraints()
+    }
+
+    func configure(for activity: Activity) {
```
Comment (ReadeP) [inline Uplift/Views/Cells/ActivityCollectionViewCell.swift:25]: **Almost all of the time, your class functions and attributes should be private**

File: Uplift/Views/Headers/ActivitesScreenHeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+//  Created by Elvis Marcelo on 4/27/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import SnapKit
+import UIKit
+
+class ActivitesScreenHeaderView: UIView {
+
+    var welcomeMessage: UILabel!
```
Comment (ReadeP) [inline Uplift/Views/Headers/ActivitesScreenHeaderView.swift:14]: same here

File: Uplift/Views/Headers/ActivitesScreenHeaderView.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    var didSetupShadow = false
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        // BACKGROUND COLOR
```
Comment (ReadeP) [inline Uplift/Views/Headers/ActivitesScreenHeaderView.swift:20]: you can delete this comment

File: Uplift/Views/Headers/ActivitesScreenHeaderView.swift [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        backgroundColor = .white
+
+        // WELCOME MESSAGE
```
Comment (ReadeP) [inline Uplift/Views/Headers/ActivitesScreenHeaderView.swift:23]: you can delete this comment

File: Uplift/Views/Home/ActivitiesListCell.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class ActivitiesListCell: ListCollectionViewCell<Activity, ActivityListItemCell> {
+
+    // MARK: - Public data vars
+    static let itemHeight: CGFloat = 110.0
```
Comment (ReadeP) [inline Uplift/Views/Home/ActivitiesListCell.swift:15]: also private

File: Uplift/Views/Home/GymsListCell.swift [hunk 8]
```diff
@@ -13,14 +13,14 @@ protocol GymsListCellDelegate: class {
 }
 
 class GymsListCell: ListCollectionViewCell<Gym, GymListItemCell> {
-
+
```
Comment (ReadeP) [inline Uplift/Views/Home/GymsListCell.swift:16]: remove tabbing

File: Uplift/Views/Cells/ActivityCollectionViewCell.swift (hunk 3 above)
Comment (emarc0314) [reply Uplift/Views/Cells/ActivityCollectionViewCell.swift:25]: configure() needs to be public to be accessed in didSelectAt by the cell

File: Uplift/Views/Home/TodaysClassesListCell.swift [hunk 9]
```diff
@@ -17,13 +17,20 @@ class TodaysClassesListCell: ListCollectionViewCell<GymClassInstance, TodaysClas
...
 
     // MARK: - Overrides
     override var config: ListConfiguration {
         return ListConfiguration(
-            itemSize: CGSize(width: 228.0, height: 195.0),
-            minimumInteritemSpacing: 16,
-            sectionInset: UIEdgeInsets(top: 0.0, left: 16.0, bottom: 32.0, right: 12.0)
+            itemSize: CGSize(width: TodaysClassesListCell.itemWidth, height: TodaysClassesListCell.itemHeight),
+            minimumInteritemSpacing: TodaysClassesListCell.minimumInterItemSpacing,
+            sectionInset: UIEdgeInsets(top: 0.0, left: TodaysClassesListCell.sectionInsetLeftRight, bottom: TodaysClassesListCell.sectionInsetBottom, right: TodaysClassesListCell.sectionInsetLeftRight)
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Uplift/Views/Home/TodaysClassesListCell.swift:33]: Look at how you did it above in lines 24-28 in `ActivitiesListCell`

File: Uplift/Views/Home/ActivitiesListCell.swift (hunk 7 above)
Comment (vinnie4k) [inline Uplift/Views/Home/ActivitiesListCell.swift:15]: Make this private

File: Uplift/Controllers/ActivitiesViewController.swift [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+extension ActivitesViewController: UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let padding: CGFloat = 10
+        return CGSize(width: (collectionView.bounds.width - padding * 2), height: 60)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 20
```
Comment (vinnie4k) [inline Uplift/Controllers/ActivitiesViewController.swift:95]: You could filter out these magic numbers to a private `Constants` struct in an extension to this class.

**Review by vinnie4k — APPROVED** [review]
Great job Elvis! Just a couple of alphabetizing that needs to be done and some small changes.

File: Uplift/Views/Home/ActivitiesListCell.swift (hunk 7 above)
Comment (emarc0314) [reply Uplift/Views/Home/ActivitiesListCell.swift:15]: I need the height of the cell in order to set the height of the collectionView as it's horizontally scrollable and I don't want it taking up more space than the cell height, so I need it to be public

**Review by ReadeP — APPROVED** [review]

**Pull request merged**